Mark helpers for constructing rich_text blocks as deprecated

jeffalder opened the pull request:
In issue #736, I called out that there are simplifying methods for constructing rich text blocks, which can't be delivered by SDK users. So that it's harder to discover this mismatch, I've removed the simplifying builders.

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [x] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/slackapi/java-slack-sdk?pullRequest=740) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/slackapi/java-slack-sdk?pullRequest=740) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/slackapi/java-slack-sdk?pullRequest=740) it.</sub>

jeffalder commented:
> First-time contributors need a maintainer to approve running workflows

seratch commented:
Hi @jeffalder, thanks for submitting this. I do understand your intention here considering the confusion you've faced but we don't remove any methods in minor releases. Until we start the next major version development (we've not decided even when to start), can we keep these methods with deprecation warnings?

jeffalder commented:
@seratch Thanks for looking. It's completely reasonable to avoid removing methods on a minor update. I've reverted the first commit and added deprecation.

Thanks again!

seratch reviewed (approved):
Thanks for updating this PR promptly! LGTM 👍
